tests-functional/clients/rpc.py

```diff
@@ -68,9 +71,22 @@ def rpc_request(self, method, params=None, request_id=None, url=None, enable_log
                 logging.debug(f"Got response: {response.content}")
         return response
 
-    def rpc_valid_request(self, method, params=None, _id=None, url=None, skip_validation=False, enable_logging=True):
-        response = self.rpc_request(method, params, _id, url, enable_logging=enable_logging)
+    def rpc_valid_request(self, method, params=None, _id=None, url=None, **kwargs):
+        skip_validation = kwargs.pop("skip_validation", False)
+        schema_check = kwargs.pop("schema_check", False)
+
+        response = self.rpc_request(method, params, _id, url, **kwargs)
         self.verify_is_valid_json_rpc_response(response, _id, skip_validation=skip_validation)
+
+        if schema_check:
+            if os.path.exists(f"{Config.base_dir}/schemas/{method}"):
+                self.verify_json_schema(response.json(), method)
+            else:
+                from utils.schema_builder import CustomSchemaBuilder
+
+                CustomSchemaBuilder(method).create_schema(response.json())
+                self.verify_json_schema(response.json(), method)
```

Comment:
I think it's better to keep the flags separate. 

1. If I understand correctly, schemas should only be generated manually and pushed to the repo. So `generate_schema` would be only temporarily enabled locally.
2. Then, if `schema_check == True`, I'd say that the schema MUST be there, so no need to check for the file.
I'd set `schema_check=True` by default, and we could then keep `schema_check=False` for specific methods.

```suggestion
        if generate_schema:
            from utils.schema_builder import CustomSchemaBuilder
            CustomSchemaBuilder(method).create_schema(response.json())

        if schema_check:
            self.verify_json_schema(response.json(), method)
```

Reply:
ok, makes sense, thanks for the suggestion!